docs/guides/git/index.md

```diff
@@ -0,0 +1,473 @@
+
+1. **~/public_html/moodle_400** containing a local installation of Moodle 4.0, and
+2. **~/public_html/moodle_master** containing a local installation of the most recent development version of Moodle.
+
+They both use your public repository at github.com as the origin. You have a branch in **moodle_400** called **MDL-xxxxx-400_topic** that was forked off **MOODLE_400_STABLE**. It contains one commit. Now, you want to apply this commit to a new branch in **moodle_master** called **MDL-xxxxx-master_topic**.
+
+```
+    cd ~/public_html/moodle_master
+    git checkout -b MDL-xxxxx-master_topic origin/master
+    git fetch ../moodle_400 MDL-xxxxx-400_topic
+    git cherry-pick FETCH_HEAD
+```
+
+1. The `git checkout -b MDL-xxxxx-master_topic origin/master` command creates new local branch, forked off **master**.
+2. The `git fetch ../moodle_400 MDL-xxxxx-400_topic` command fetches all data needed to apply the topic branch, and sets the pointer to the tip of that branch with **FETCH_HEAD** as a symbolic reference.
+3. The `git cherry-pick FETCH_HEAD` command picks the tip of the branch (the top-most commit) and tries to apply it on the current branch.
+
+:::tip
+There's a variant of the `git cherry-pick` command that supports multiple commits. Use `$ git cherry-pick A^..B` if you want to include commits from "A" to "B" (inclusive). Commit "A" should be older than "B".
+:::
+
+### Applying a set of patches
+<!-- cspell:ignore patchset -->
+Consider the branch **MDL-xxxxx-400_topic** from the previous example. Imagine the branch consisted of several commits. It may be easier to use the `git format-patch` and `git am` commands to apply the whole set of patches (patchset).
+
+Firstly, you will need to export all commits from the topic branch to separate files.
+
+```
+    cd ~/public_html/moodle_400
+    mkdir .patches
+    git format-patch -o .patches MOODLE_400_STABLE..MDL-xxxxx-400_topic
+```
+
+The `git format-patch -o .patches MOODLE_400_STABLE..MDL-xxxxx-400_topic` command takes all commits from the topic branch that are not in **MOODLE_400_STABLE** and exports them, one by one, to the output directory **.patches**. The generated files will contain the patch itself (in diff format), and additional information about the commit. This format allows you to easily email the files to another person for peer-review.
+
+In this example, we will apply them to another repository.
+
+```
+    cd ~/public_html/moodle_master
+    git checkout -b MDL-xxxxx-master_topic origin/master
+    git am -3 ../moodle_400/.patches/*
+```
+
+The `git am -3 ../moodle_400/.patches/*` command applies all the files from the **.patches** directory. When a patch does not apply cleanly, the command tries to fallback on a three-way merge (see the -3 parameter). If conflicts occur during the procedure, you can resolve them and use `git am --continue`, or abort the whole procedure with `git am --abort`.
+
+## <a name="see_also"></a> See also
```

Comment:
See docs here: https://docusaurus.io/docs/next/markdown-features/toc#heading-ids

```suggestion
## See also {#see_also}
```

But I don't know if we really need/want to customise this.